build-support/pants_venv

```diff
@@ -11,26 +11,52 @@ REQUIREMENTS=(
   ${REPO_ROOT}/pants-plugins/3rdparty/python/requirements.txt
 )
 
+venv_dir_prefix="${REPO_ROOT}/build-support/pants_dev_deps"
+
+function determine_specific_py3_version() {
+  version_number="$(${PY} -c 'import sys; print(sys.version_info[1])')"
+  case "${version_number}" in
+    6) echo "36" ;;
+    7) echo "37" ;;
+    8) echo "38" ;;
+    *) die "Pants cannot run with Python 3.${version_number}." ;;
+  esac
+}
+
 function venv_dir() {
-  py_version="py2"; [ "${PANTS_USE_PYTHON3}" = true ] && py_version="py3"
-  echo ${REPO_ROOT}/build-support/pants_dev_deps.${py_version}.venv
+  case "${PY}" in
+    'python3') py_venv_version="$(determine_specific_py3_version)" ;;
+    'python3.6') py_venv_version="36" ;;
+    'python3.7') py_venv_version="37" ;;
+    'python3.8') py_venv_version="38" ;;
+    # If $PY not pre-set by caller, default to Python 2.7
+    *) py_venv_version="27"
```

Comment:
I couldn't think of a way to make this switch statement more generic. We must specify the specific minor version in order to make the renamed `venv` folder solution work, which is necessary to solve problem 1 from the PR description.

Let me know if you have any suggestions.